app/src/main/java/org/p2p/wallet/solend/ui/deposit/SolendDepositPresenter.kt

```diff
@@ -30,14 +28,12 @@ class SolendDepositPresenter(
         if (validDeposits.isEmpty()) {
             launch {
                 try {
-                    validDeposits = solendDepositsInteractor.getUserDeposits()
-                        .filter { depositToken ->
-                            userInteractor.getUserTokens()
-                                .firstOrNull { token -> token.tokenSymbol == depositToken.tokenSymbol }
-                                ?.totalInUsd
-                                .orZero()
-                                .isNotZero()
-                        }
+                    val userTokens = userInteractor.getUserTokens()
+                    validDeposits = solendDepositsInteractor.getUserDeposits().onEach { depositToken ->
+                        depositToken.availableTokensForDeposit = userTokens.firstOrNull { token ->
```

Comment:
i suggest to do copy() instead of var

Reply:
and maybe fill availableTokensForDeposit on the interactor level or repository level

if the presenter and domain requires availableTokensForDeposit to be filled, it's the repository responsibility to do that in the mapper

Reply:
we need to move it up on root of feature level